15652522 SUNBT6964562 xscreensaver dumps core whenever it unlocks in svn-142

Upstream status unknown - suspect to be specific to our GTK unlock dialog.
---
 driver/lock.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/driver/lock.c b/driver/lock.c
--- a/driver/lock.c
+++ b/driver/lock.c
@@ -3092,7 +3092,7 @@ auth_finished_cb (saver_info *si)
   }
 
  END:
-  if (si->pw_data)
+  if (si->pw_data && si->pw_data->prompt_screen)
     destroy_passwd_window (si);
 }
